guides/asynchronous-tasks/readme.md

```diff
@@ -264,7 +266,7 @@ As tasks run synchronously until they yield back to the reactor, you can guarant
 
 ## Exception Handling
 
-{ruby Async::Task} captures and logs exceptions. All unhandled exceptions will cause the enclosing task to enter the `:failed` state. Non-`StandardError` exceptions are re-raised immediately and will generally cause the reactor to fail. This ensures that exceptions will always be visible and cause the program to fail appropriately.
+{ruby Async::Task} captures and logs exceptions. All unhandled `StandardError`-derived exceptions will cause the enclosing task to enter the `:failed` state. Non-`StandardError` exceptions are re-raised from the reactor and thus cause it to fail. This ensures that exceptions will always be visible and cause the program to fail appropriately.
```

Comment:
```suggestion
{ruby Async::Task} captures and logs exceptions. All unhandled exceptions will cause the enclosing task to enter the `:failed` state. Non-`StandardError` exceptions are re-raised immediately and will cause the reactor to exit. This ensures that exceptions will always be visible and cause the program to fail appropriately.
```

I suppose we could go with this. The exceptions are not reprised from the reactor - they are reprised from the fiber which generally causes the reactor to exit if it was the one that caused resume. Internally, it's more complex.